### ISSUE TITLE
Spectator mode cannot move if gamerule render chunks set to false even in loaded chunks

### ISSUE THREAD

**Issue by JMCY26** [PaperMC/Paper#2766]
### What behaviour is expected:
Never had this issue before with paper so I believe for it to be recent.
When /gamerule spectatorGenerateChunks is set to false, trying to move around in Spectator mode, you cannot leave the chunk you are in even if it is loaded by another player in survival. We've used this game rule for a long time and haven't changed any paper configuration since updating.

### What behaviour is observed:
Camera keeps jumping back jittery, not allowing to move around.

### Steps/models to reproduce:
Change spectator gamerule, jump in spectator mode and try move in loaded chunks.

### Plugin list:
Coreprotect, DiscordSRV, Essentials, LuckPerms, Vault

### Paper build number:
Paper-19

### Anything else:
NA

**Comment by aeonlamb** [comment]
I was going to report this bug myself before, because I noticed it in paper 1.14.4 as well. However, I decided not to because it is caused by a vanilla bug. I held out hope mojang would notice it and patch it before 1.15 release, but looks like they didn't:

[https://bugs.mojang.com/browse/MC-157812](https://bugs.mojang.com/browse/MC-157812)

The bug here is that vanilla will treat "spectatorsGenerateChunks" gamerule as if spectators cannot even cause chunks to remain loaded. With this gamerule, spectators can no longer even load chunks, let alone generate them (which is a behavior change from 1.13.x and below).

However, the "camera stuttering" and being able to even move into other chunks is due to a paper config setting: "prevent-moving-into-unloaded-chunks". If you have this config set to true, you can't move into chunks in spectator mode with this gamerule enabled because the chunks are no longer loaded.

So basically, it's mojang's fault, but if you want to fix it you can simply either disable this gamerule OR change the paper config until it is fixed. Personally, I just disabled the gamerule for now.

**Comment by Spottedleaf** [comment]
The code that controls spectator chunk loading (specifically, general player chunk loading) has not changed since 1.14, so I doubt you have issues that weren't present in 1.14. Check again
It's very clear from the code that this is the behaviour:

If the gamerule is false, then spectators do not add player tickets to chunks. This means that there is no attempt to load the chunks around the player, whether or not they are generated.

If the gamerule is true, then spectators do add player tickets to chunks, like how normal players do.

All players, regardless of whether they are in spectator and the gamerule, will be sent chunks within the server view distance.

**Comment by JMCY26** [comment]
Changing paper config to allow players to move in unloaded chunks worked, thanks.
